index.bs

```diff
@@ -74,6 +75,125 @@ A user agent must not <a spec=HTML>delay the load event</a> of the <{link}> elem
 
 <p class="issue">TODO: We should define a list of allowable referrer policies to prerender with, and early-return if the above referrer policy is not in this list.
 
+<h2 id="speculation-rules">Speculation rules</h2>
+
+<h3 id="speculation-rules-dfns">Definitions</h3>
+
+A <dfn>speculation rule</dfn> is a [=struct=] with the following [=struct/items=]:
+* <dfn for="speculation rule">URLs</dfn>, an [=ordered set=] of [=URLs=]
+* <dfn for="speculation rule">requirements</dfn>, an [=ordered set=] of [=strings=]
+
+The only valid string for [=speculation rule/requirements=] to contain is "`anonymous-client-ip`".
+
+A <dfn>speculation rule set</dfn> is a [=struct=] with the following [=struct/items=]:
+* <dfn for="speculation rule set">prefetch rules</dfn>, a [=list=] of [=speculation rules=]
+
+<h3 id="speculation-rules-script">The <{script}> element</h3>
+
+<em>Note</em>: This section contains modifications to the corresponding section of [[HTML]].
+
+<p class="issue">We should consider whether we also want to make this execute even if scripting is disabled.</p>
+
+<p class="issue">We should also incorporate the case where a {{HTMLScriptElement/src}} attribute is set.</p>
+
+<p class="issue">We could fire {{HTMLElement/error}} and {{HTMLElement/load}} events if we wanted to.</p>
+
+Add "`speculationrules`" to the list of valid values for <a spec=html>the script's type</a>.
+
+Rename [=the script's script=] to <dfn>the script's result</dfn>, which can be a [=speculation rule set=] in addition to any other valid type.
+
+Inside the [=prepare a script=] algorithm:
+
+* Insert the following step as the second-last sub-step under "Determine the script's type as follows:":
+  * If the script block's type string is an [=ASCII case-insensitive=] match for the string "`speculationrules`", <a spec=html>the script's type</a> is "`speculationrules`".
+
+* Insert the following case in the switch on <a spec=html>the script's type</a> within the step which begins "If the element does not have a {{HTMLScriptElement/src}} content attribute..."
+  <dl>
+    <dt>"`speculationrules`"</dt>
+    <dd>
+      1. Let |result| be the result of [=parse speculation rules|parsing speculation rules=] given source text and base URL.
+
+      1. If |result| is not null, set [=the script's result=] to |result|.
+
+      1. <a spec=html>The script is ready</a>.
+    </dd>
+  </dl>
+
+Inside the <a spec=html>execute a script block</a> algorithm:
+
+<p class="issue">Adjust this consistent with the resolution of <a href="https://github.com/WICG/import-maps/issues/243">WICG/import-maps#243</a>.</p>
```

Comment:
As per https://github.com/WICG/import-maps/issues/243#issuecomment-764452366, I think we can remove this issue but I believe we'll need to add some readiness handling like https://wicg.github.io/import-maps/#ref-for-concept-script-type%E2%91%A2.

If we don't "handle" this script type in [prepare a script](https://html.spec.whatwg.org/multipage/scripting.html#prepare-a-script) step 28, then the "Otherwise" block will trigger, thus executing the script immediately even if other scripts are running, which seems particularly bad here. So we should have some sort of "type=importmap" handling here. Exactly what that handling will do ties into a comment I have below, so please continue reading there.